share/build/buildlib.spio

```diff
@@ -86,80 +92,95 @@ def buildlib(bldroot, installpath, case):
     # also defined in the environment.  In this case we
     # will use the installed pio and not build it here.
     installed_pio_version = os.environ.get("PIO_VERSION_MAJOR")
-    logger.info(
-        "pio_version_major = {} pio_version = {}".format(
-            installed_pio_version, pio_version
-        )
-    )
+    logger.info(f"pio_version_major = {installed_pio_version} pio_version = {pio_version}")
     if installed_pio_version is not None and int(installed_pio_version) == pio_version:
         logger.info("Using installed PIO library")
         _set_pio_valid_values(case, os.environ.get("PIO_TYPENAME_VALID_VALUES"))
         return
 
     pio_model = "pio{}".format(pio_version)
-    pio_dir = os.path.join(bldroot, pio_model)
-    if not os.path.isdir(pio_dir):
-        os.makedirs(pio_dir)
-    casetools = case.get_value("CASETOOLS")
-    if scorpio_src_root_dir:
-        # Use old genf90 until "short" type is supported
-        cmake_opts = (
-            '"-D GENF90_PATH='
-            + os.path.join(scorpio_src_root_dir, scorpio_dir, "src/genf90")
-            + '" '
-        )
-    elif pio_version == 1:
-        cmake_opts = '"-D GENF90_PATH=$CIMEROOT/CIME/non_py/externals/genf90 "'
+    pio_bld_dir = os.path.join(bldroot, pio_model)
+    if not os.path.isdir(pio_bld_dir):
+        os.makedirs(pio_bld_dir)
+
+    # Compute cmake args
+    # Use old genf90 until "short" type is supported
+    cmake_opts = "-Wno-dev "
+    cmake_opts += f"-DGENF90_PATH={scorpio_src_dir}/src/genf90 "
+    if "ADIOS2_ROOT" in os.environ:
+        cmake_opts += "-DWITH_ADIOS2:BOOL=ON "
+    if debug:
+        cmake_opts += "-DPIO_ENABLE_LOGGING=ON "
+    # Case changes for NetCDF/NETCDF forces us to do this. For other packages
+    # what's already in the env should work
+    if "NETCDF_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_PATH:PATH={os.environ['NETCDF_PATH']} "
+    if "NETCDF_C_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_C_PATH:PATH={os.environ['NETCDF_C_PATH']} "
+    if "NETCDF_FORTRAN_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_Fortran_PATH:PATH={os.environ['NETCDF_FORTRAN_PATH']} "
+
+    # scorpio needs a little help finding hdf5. The FindHDF5 in scopio uses the
+    # env var "HDF5" instead of "HDF5_ROOT". If HDF5_ROOT is not set, but we have
+    # h5dump in our path, we can use that info to set up HDF5.
+    which_h5dump = shutil.which('h5dump')
```

Comment:
@jayeshkrishna , @dqwu, is this approach too aggressive? It means any environment that has h5dump in the PATH will result in WITH_HDF5=On for scorpio.

Reply:
I think we do not have to enable HDF5 support by default so far, at least before scorpio 1.5.0 becomes the default sub module.

Reply:
OK, I rolled this back.